native_client/ctcdecode/ctc_beam_search_decoder.cpp

```diff
@@ -96,24 +96,52 @@ DecoderState::next(const double *probs,
         if (full_beam && log_prob_c + prefix->score < min_cutoff) {
           break;
         }
+        if (prefix->score == -NUM_FLT_INF) {
+          continue;
+        }
+		if (!prefix->is_empty() && prefix->timesteps.empty()) {
+		  // This should never happen. But we report it if it does.
+		  std::cerr<<"error: non-empty prefix has empty timestep sequence"<<std::endl;
+		  continue;
```

Comment:
If this should never happen we should just assert it instead of printing and continuing. That way we find out about problems early.

Reply:
ok, sure